File: tools/sam/data/bed.py
# -*- coding: utf-8 -*-

#
#   ngsv
#   http://github.com/xcoo/ngsv
#   Copyright (C) 2012, Xcoo, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

from sql import *

import json


class Bed:

    def __init__(self, db):
        self.db = db

    def get_by_filename(self, filename):

        sql = u"SELECT bed_id, file_name, created_date, item_rgb FROM bed WHERE file_name = '%s'" % escape(filename)

        result = self.db.execute(sql)

        if len(result) > 0:
            return {'id': result[0][0],
                    'name': result[0][1],
                    'created': result[0][2],
                    'rgb': result[0][3]}
        else:
            return None

    def append(self, filename,
               trackname, description, visibility, itemRgb):

        SQL_TEMPLATE = u"INSERT INTO bed (file_name, created_date, track_name, description, visibility, item_rgb) VALUES ('%s', %d, '%s', '%s', %d, %d)"

        sql = SQL_TEMPLATE % (escape(filename),
                              get_time(),
                              escape(json.dumps(trackname)),
                              escape(json.dumps(description)),
                              visibility,
                              itemRgb)

        self.db.execute(sql)

File: main/ngsView/src/main/java/genome/data/Bed.java
/*
 *   ngs View
 *   http://casmi.github.com/
 *   Copyright (C) 2011-2012, Xcoo, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package genome.data;

import casmi.sql.Entity;
import casmi.sql.annotation.Fieldname;
import casmi.sql.annotation.Ignore;
import casmi.sql.annotation.PrimaryKey;
import casmi.sql.annotation.Tablename;

@Tablename("bed")
public class Bed extends Entity {
	@PrimaryKey
	private long bedId;

	@Fieldname("fileName")
	private String fileName;

	@Fieldname("createdDate")
	private long createdDate;

	@Ignore
    private String trackName;

	@Ignore
    private String description;

	@Ignore
    private int visibility;
	
	@Ignore
	private int itemRgb;

	@Ignore
	private boolean selected;
	
	@Ignore
	private BedFragment[] bedFragments;
	
	public long getBedId() {
		return bedId;
	}

	public void setBedId(long bedId) {
		this.bedId = bedId;
	}

	public String getFileName() {
		return fileName;
	}

	public void setFileName(String fileName) {
		this.fileName = fileName;
	}

	public long getCreatedDate() {
		return createdDate;
	}

	public void setCreatedDate(long createdDate) {
		this.createdDate = createdDate;
	}

	public String getTrackName() {
		return trackName;
	}

	public void setTrackName(String trackName) {
		this.trackName = trackName;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public int getVisibility() {
		return visibility;
	}

	public void setVisibility(int visibility) {
		this.visibility = visibility;
	}

	public int getItemRgb() {
		return itemRgb;
	}

	public void setItemRgb(int itemRgb) {
		this.itemRgb = itemRgb;
	}

	public boolean isSelected() {
		return selected;
	}

	public void setSelected(boolean selected) {
		this.selected = selected;
	}
    
    public BedFragment[] getBedFragments() {
        return bedFragments;
    }
    
    public void setBedFragments(BedFragment[] bedFragments) {
        this.bedFragments = bedFragments;
    }

}
